framework/src/modules/random/stores/used_hash_onions.ts

```diff
@@ -62,10 +58,18 @@ export class UsedHashOnionsStore extends BaseOffchainStore<UsedHashOnionStoreObj
 		address: Buffer,
 		usedHashOnionInfo: UsedHashOnionInfo,
 	): Promise<void> {
-		const UsedHashOnionStoreObject = await this.get(ctx, address);
+		const usedHashOnionStoreObject = await this.get(ctx, address);
 
-		UsedHashOnionStoreObject.usedHashOnions.usedHashOnions.push(usedHashOnionInfo);
+		const index = usedHashOnionStoreObject.usedHashOnions.usedHashOnions.findIndex(
+			ho => ho.count === usedHashOnionInfo.count,
+		);
+
+		if (index === -1) {
```

Comment:
The changes now update if the usedHashOnion with same count exists, could you please confirm or correct the understanding?